app/services/import_notifier.rb

```diff
@@ -4,15 +4,41 @@ def initialize(description: nil)
   end
 
   def meters_running_behind
-    meters = Meter.active
-    .joins(:amr_data_feed_readings)
-    .where("amr_data_feed_readings.created_at >= NOW() - INTERVAL '1 year'")
-    .joins("INNER JOIN amr_data_feed_configs on amr_data_feed_readings.amr_data_feed_config_id = amr_data_feed_configs.id")
-    .where.not("amr_data_feed_configs.import_warning_days" => nil)
-    .joins(:amr_validated_readings)
-    .group('meters.id')
-    .having("MAX(amr_validated_readings.reading_date) < NOW() - MIN(amr_data_feed_configs.import_warning_days) * '1 day'::interval")
-    meters.sort_by {|m| [m.school.area_name, m.meter_type, m.school_name, m.mpan_mprn]}
+    find_meters_running_behind.sort_by do |meter|
+      [
+        meter&.school&.area_name || '',
+        meter&.meter_type || '',
+        meter&.school_name || '',
+        meter&.mpan_mprn || 0
```

Comment:
Yes all the columns are required.

Agree it would be better to sort in the database, but we can add that separately. What's the reason for doing the defaults in the (e.g. `|| 0`) as the current version doesn't use those and all those fields are required I think?